pkg/ast/pipesearch/searchHandler.go

```diff
@@ -527,41 +530,64 @@ func convertQueryCountToTotalResponse(qc *structs.QueryCount) interface{} {
 }
 
 /*
+Function designed to parse Time from a String input.
    Supports "now-[Num][Unit]"
    Num ==> any positive integer
    Unit ==> m(minutes), h(hours), d(days)
 */
 
-func parseAlphaNumTime(nowTs uint64, inp string, defValue uint64) uint64 {
+func parseAlphaNumTime(nowTs uint64, inp string, defValue uint64) (uint64, error) {
 
+	//removing spaces if any from input
 	sanTime := strings.ReplaceAll(inp, " ", "")
 
+	//to be returned as is if no filters are applied
 	if sanTime == "now" {
-		return nowTs
+		return nowTs, nil
 	}
 
+	//setting return value as default
 	retVal := defValue
 
+	//regex pattern for "now-[number]" and unit at the end which only accepts 'm', 'h' and 'd'
+	pattern := `^now-\d+[mhd]$`
+	r, patternError := regexp.Compile(pattern)
+	if patternError != nil {
+		log.Errorf("startEpoch/endEpoch Input received is invalid. Value probably modified")
+		return retVal, fmt.Errorf("startEpoch Input received is invalid.")
+	}
+
 	strln := len(sanTime)
-	if strln < 6 {
-		return retVal
+	patternMatch := r.MatchString(sanTime)
+	if !patternMatch {
+		//impossible case reached, return error
+		log.Errorf("startEpoch/endEpoch Input received is invalid. Value probably modified")
+		return retVal, fmt.Errorf("startEpoch Input received is invalid.")
 	}
 
+	//extracting Unit from the string
 	unit := sanTime[strln-1]
 	num, err := strconv.ParseInt(sanTime[4:strln-1], 0, 64)
 	if err != nil {
-		return defValue
+		log.Errorf("startEpoch/endEpoch Printing error below")
+		log.Errorf(err.Error())
+		return defValue, fmt.Errorf("startEpoch Input received is invalid.")
 	}
```

Comment:
Again, don't mention the `start/end epoch` since this function doesn't know about that. You can also remove the error log on line 572. For the one on 573, prepend it with the name of this function so it's easier to track down where this error came from if we see it in our log file when the program is running

Similar comment for lines 585-588

Reply:
Got it